src/LivePreviewRenderer.ts

```diff
@@ -0,0 +1,65 @@
+import { EditorView, PluginValue, ViewPlugin } from '@codemirror/view';
+
+import { Task } from './Task';
+
+export const newLivePreviewRenderer = () => {
+    return ViewPlugin.fromClass(LivePreviewRenderer);
+};
+
+class LivePreviewRenderer implements PluginValue {
+    private readonly view: EditorView;
+
+    constructor(view: EditorView) {
+        this.view = view;
+        this.view.dom.onClickEvent(this.handleClickEvent.bind(this));
+    }
+
+    public update(): void {}
+
+    private handleClickEvent(event: MouseEvent): boolean {
+        // Only handle checkbox clicks.
+        if (!event.target || (event.target as any).type !== 'checkbox') {
+            return false;
+        }
+
+        const { target } = event;
+        const { state } = this.view;
+        const position = this.view.posAtDOM(target as Node);
+        const line = state.doc.lineAt(position);
+        const task = Task.fromLine({
+            line: line.text,
+            path: '',
+            sectionStart: 0,
+            sectionIndex: 0,
+            precedingHeader: null,
+        });
+
+        // Only handle checkboxes of tasks.
+        if (task === null) {
+            return false;
+        }
+
+        // We need to prevent default so that the checkbox is only handled by us and not obsidian.
+        event.preventDefault();
+        // We need to stop immediate propagation in order for the event to only trigger once.
+        event.stopImmediatePropagation();
+
+        // Clicked on a task's checkbox. Toggle the task and set it.
+        const toggled = task.toggle();
+        const toggledString = toggled
+            .map((task) => task.toFileLineString())
+            .join(state.lineBreak);
+
+        // Creates a CodeMirror transaction in order to updat the document.
+        const transaction = state.update({
```

Comment:
I need to replace more than a single character. That's the whole point of this class :smile: It could be required to append or remove a done date. Or it may be required to add a new line above with the new instance of a recurring task.